src/main/java/com/microsoft/azure/util/AzureCredentials.java

```diff
@@ -530,7 +530,14 @@ public static TokenCredential getSystemCredentialById(String credentialID) {
                 CredentialsMatchers.withId(credentialID));
 
         if (!azureImdsCredentials.isEmpty()) {
-            return new ManagedIdentityCredentialBuilder().build();
+            AzureImdsCredentials azureIdmsCredential = azureImdsCredentials.get(0);
+            ManagedIdentityCredentialBuilder credentialBuilder = new ManagedIdentityCredentialBuilder();
+
+            if (null != azureIdmsCredential.getClientId()) {
```

Comment:
Use `Util.fixEmpty`, as UI submissions will submit an empty string

https://javadoc.jenkins.io/hudson/Util.html#fixEmpty(java.lang.String)

```suggestion
            if (Util.fixEmpty(azureIdmsCredential.getClientId()) != null) {
```

Reply:
Should I include the call to Util.fixEmpty directly in the getter method ? (So it doesn't have to be repeater outside the class)

Reply:
sure can we avoid the [yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) then and use the format

```suggestion
            if (azureIdmsCredential.getClientId() != null) {
```

Reply:
Done